components/brave_wallet/browser/meld_integration_service_unittest.cc

```diff
@@ -0,0 +1,1019 @@
+      const std::string& service_providers,
+      const std::string& payment_method_types,
+      const std::string& statuses,
+      MeldIntegrationService::GetServiceProvidersCallback callback,
+      const net::HttpStatusCode http_status = net::HTTP_OK) {
+    SetInterceptor(content, http_status);
+
+    base::RunLoop run_loop;
+    auto filter = mojom::MeldFilter::New(countries, fiat_currencies,
+                                         crypto_currencies, service_providers,
+                                         payment_method_types, statuses);
+
+    asset_ratio_service_->GetServiceProviders(
+        std::move(filter),
+        base::BindLambdaForTesting(
+            [&](std::optional<std::vector<mojom::MeldServiceProviderPtr>> sps,
+                const std::optional<std::vector<std::string>>& errors) {
+              std::move(callback).Run(std::move(sps), errors);
+              run_loop.Quit();
+            }));
+    run_loop.Run();
+  }
+
+  void TestGetCryptoQuotes(
+      const std::string& content,
+      const std::string& country,
+      const std::string& from_asset,
+      const std::string& to_asset,
+      const double source_amount,
+      const std::string& account,
+      MeldIntegrationService::GetCryptoQuotesCallback callback,
+      const net::HttpStatusCode http_status = net::HTTP_OK) {
+    SetInterceptor(content, http_status);
+
+    base::RunLoop run_loop;
+    asset_ratio_service_->GetCryptoQuotes(
+        country, from_asset, to_asset, source_amount, account,
+        base::BindLambdaForTesting(
+            [&](std::optional<std::vector<mojom::MeldCryptoQuotePtr>> quotes,
+                const std::optional<std::vector<std::string>>& errors) {
+              std::move(callback).Run(std::move(quotes), errors);
+              run_loop.Quit();
+            }));
+    run_loop.Run();
+  }
+
+  void TestGetPaymentMethods(
+      const std::string& content,
+      const std::string& countries,
+      const std::string& fiat_currencies,
+      const std::string& crypto_currencies,
+      const std::string& service_providers,
+      const std::string& payment_method_types,
+      const std::string& statuses,
+      MeldIntegrationService::GetPaymentMethodsCallback callback,
+      const net::HttpStatusCode http_status = net::HTTP_OK) {
+    SetInterceptor(content, http_status);
+
+    base::RunLoop run_loop;
+    auto filter = mojom::MeldFilter::New(countries, fiat_currencies,
+                                         crypto_currencies, service_providers,
+                                         payment_method_types, statuses);
+
+    asset_ratio_service_->GetPaymentMethods(
+        std::move(filter),
+        base::BindLambdaForTesting(
+            [&](std::optional<std::vector<mojom::MeldPaymentMethodPtr>>
+                    payment_methods,
+                const std::optional<std::vector<std::string>>& errors) {
+              std::move(callback).Run(std::move(payment_methods), errors);
```

Comment:
Are we testing that callback is being executed? maybe use MockCallback instead?

Reply:
No, we pass to the TestGetServiceProvider a callback, where we check is all data parsed correctly.
We need to do it instead of Mock, because only in this way we can go through all parsing steps (json sanitizing and parsing)
and then check the correctness of the parsed data in the callback.